uaclient/security.py

```diff
@@ -485,28 +487,74 @@ def prompt_for_affected_packages(
         return
     upgrade_packages = []  # Packages from Ubuntu standard updates
     upgrade_packages_ua = []  # Packages requiring UA subscription
-    for pkg_num, pkg_name in enumerate(sorted(affected_pkg_status.keys()), 1):
-        print("({}/{}) {}:".format(pkg_num, count, pkg_name))
-        pkg_status = affected_pkg_status[pkg_name]
-        print(pkg_status.status_message)
-        if pkg_status.status == "released":
-            update_needed = False
-            for binary_pkg, version in installed_packages[pkg_name].items():
-                if not version_cmp_le(pkg_status.fixed_version, version):
-                    update_needed = True
-                    if pkg_status.requires_ua:
-                        upgrade_packages_ua.append(binary_pkg)
-                    else:
-                        upgrade_packages.append(binary_pkg)
-            if update_needed:
-                print(status.MESSAGE_SECURITY_UPDATE_NOT_INSTALLED)
+    is_fixable = True
+    pkg_index = 0
+
+    for pkg_status, pkg_status_group in groupby(
+        sorted(affected_pkg_status.items(), key=lambda x: (x[1].status, x[0])),
+        key=lambda x: x[1].status.replace("ignored", "deferred"),
+    ):
+        if pkg_status != "released":
```

Comment:
This `prompt_for_affected_packages` is not getting quite long and hard to follow. I would like us to trim down into more discrete helper functions, but I know my existing PR is colliding with this section of code as well and trying to break out those separate functions will make the current merge conflicts a bit harder I think. So, let's carry the todo item after this PR and #1446 land that we make sure `prompt_for_affected_packages` is a more readable function will less operations being performed.

Reply:
I completely agree here, we should refactor that function after we land #1446